rs/execution_environment/src/scheduler.rs

```diff
@@ -683,6 +691,10 @@ impl SchedulerImpl {
             instructions: round_limits.instructions,
             subnet_available_memory: (round_limits.subnet_available_memory
                 / self.config.scheduler_cores as i64),
+            // This is a soft cap, it is unnecessary to strictly divide the pool among
+            // the threads. If it is exceeded, canisters can still rely on their guaranteed
+            // callback quota.
+            subnet_available_callbacks: round_limits.subnet_available_callbacks,
```

Comment:
I think we implicitly rely on the fact that execution can only create a relatively small number of messages per round -- much smaller than `subnet_available_callbacks` limit. Perhaps also something worth doublechecking with execution.

Reply:
Yes, I believe we are. The initial limits we discussed in the design doc were a guaranteed quota of 50 callbacks plus a shared pool of 5M callbacks. This soft limit of 1M callbacks means a worst-case 4M callbacks (assuming that we start off with zero and that each of 4 threads can make 1M calls within a round; which I'm not sure is even possible).

I prefer the soft cap instead of a hard limit because (i) it allows you to take shortcuts such as this one instead of being forced into splitting the resources equally between threads; and (ii) it still ends up with a lower overall limit in this case.